src/clojurians_log/views.clj

```diff
@@ -280,6 +281,25 @@
      [:section#about
       (:data/about-hiccup context)]]]])
 
+(defn- user-profile-html [context]
+  [:html.user-profile-page
+   [:body 
+    [:h2 {:style "text-align:center; margin-top: 20px; font-family: sans-serif;"} "User Profile"]
+    [:div {:style " box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2); max-width: 350px; 
+                   margin: auto; text-align: center; font-family: sans-serif;"}
+
+     (for [[k v] (select-keys (get-in context [:data/username 0])
+                              [:user-profile/display-name :user-profile/real-name :user-profile/image-192 :user/slack-id])
+           :when (-> k)]
+       (case k
+         :user-profile/image-192 [:img {:src [v], :alt "John", :style "width:95%; border-radius:80%; padding: 8px 0px;"}]
+         :user-profile/display-name [:h1 [k v]]
+         :user-profile/real-name [:p {:style "color: grey; font-size: 18px;"} [k v]]
+         :user/slack-id [:a {:style "border: none; outline: 0; display: inline-block; padding: 8px 0px; 
+                                    color: white; background-color: #000; text-align: center; cursor: pointer; 
+                                    width: 100%; font-size: 18px;"
+                             :href (str slack-instance "/team/" v )} "let's talk on Slack"]))]]])
```

Comment:
Quite an original solution! I guess the `select-keys` also ensures the ordering? I would do this a lot less clever.

```
(let [{:user-profile/keys [display-name real-name image-192] :user/keys [slack-id]} (get-in context [:data/username 0])]
  (list
    (when image-192
      [:img {:src image-192} ...])
    (when display-name
      [:h1 display-name])
    ...
```

Some other notes here
- `:alt "John"` I guess not everyone is called John :)
- `:src [v]` what's that vector doing? Just `:src v` should be fine
- `[:h1 [k v]]` that inner vector is definitely not right, this will end up in html as `<h1><user-profile/display-name>...</user-profile/display-name></h1>`
- inline styling -> styles.clj :see_no_evil: 
- Why is it `:data/username` when what's in there is a whole bunch of user profile information? shouldn't it then be `:data/user-profile`? Why the `0` in the `get-in`?